### PR TITLE
feat: Improve the `data_structures/reverse_a_linked_list.cpp` file

### PR DESCRIPTION
**Pull request by ayushkedia05**
instead of returning int_32 we can return a bool value and the interger which is to be compared in top,last and be passed as a argument. this is the better way of removing the bugs from the code. [BUG] Fix CodeQL warnings this issue can be easily solved using this method.

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1838"></a>

### CODE REVIEW

File: data_structures/reverse_a_linked_list.cpp [hunk 1]
```diff
@@ -115,61 +86,65 @@ void list::reverseList() {
...
         }
         count++;
         current = current->next;
     }
-
     /* if we get to this line,the caller was asking for a non-existent element
     so we assert fail */
-    assert(0);
+    return false;
+// Function traverse should return a value of type int32_t but does not return a value here
```
Comment (Panquesito7) [inline data_structures/reverse_a_linked_list.cpp:144]: Could you please adjust this comment to be in line 141 along with the current comment in that line?

File: data_structures/reverse_a_linked_list.cpp [hunk 2]
```diff
@@ -1,32 +1,7 @@
...
- * to the next of current, we store it in the next_node pointer, now we store
- * the prev pointer in the current of next, this is where the actual reversal
- * happens. And then we move the prev and current pointers one step forward.
- * Then the head node is made to point to the last node (prev pointer) after
- * completion of an iteration.
-
- * [A graphic explanation and view of what's happening behind the
- *scenes](https://drive.google.com/file/d/1pM5COF0wx-wermnNy_svtyZquaCUP2xS/view?usp=sharing)
- */
-
```
Comment (Panquesito7) [inline data_structures/reverse_a_linked_list.cpp:24]: Please add this documentation back.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: data_structures/reverse_a_linked_list.cpp (hunk 1 above)
Comment (ayushkedia05) [reply data_structures/reverse_a_linked_list.cpp]: done

File: data_structures/reverse_a_linked_list.cpp (hunk 2 above)
Comment (ayushkedia05) [reply data_structures/reverse_a_linked_list.cpp]: done

File: data_structures/reverse_a_linked_list.cpp [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Panquesito7) [inline data_structures/reverse_a_linked_list.cpp:1]:
```suggestion
```

File: data_structures/reverse_a_linked_list.cpp [hunk 4]
```diff
@@ -17,16 +18,18 @@
...
  * completion of an iteration.
-
  * [A graphic explanation and view of what's happening behind the
  *scenes](https://drive.google.com/file/d/1pM5COF0wx-wermnNy_svtyZquaCUP2xS/view?usp=sharing)
  */
 
+
+
+
+
```
Comment (Panquesito7) [inline data_structures/reverse_a_linked_list.cpp:28]:
```suggestion
```

File: data_structures/reverse_a_linked_list.cpp [hunk 5]
```diff
@@ -115,61 +114,64 @@ void list::reverseList() {
...
             Node *t = head;
             while (t->next != nullptr) {
                 t = t->next;
             }
-            return t->val;
+             return (t->val==checkvalue)?true:false;
         }
     } catch (const std::exception &e) {
         std::cerr << "List is empty" << e.what() << '\n';
+// returning false value is item is not found i.e lisked is empty
```
Comment (Panquesito7) [inline data_structures/reverse_a_linked_list.cpp:149]: Please fix the indentation (same in other comments).

**Review by Panquesito7 — COMMENTED** [review]
Almost there! 😄

File: data_structures/reverse_a_linked_list.cpp (hunk 3 above)
Comment (Panquesito7) [inline data_structures/reverse_a_linked_list.cpp:1]:
```suggestion
```

**Pull request closed without merging**